src/coreclr/nativeaot/Runtime/FinalizerHelpers.cpp

```diff
@@ -135,7 +135,10 @@ EXTERN_C void QCALLTYPE RhWaitForPendingFinalizers(UInt32_BOOL allowReentrantWai
         // Wait for the finalizer thread to get back to us.
         g_FinalizerDoneEvent.Wait(INFINITE, false, allowReentrantWait);
 
-        if (desiredFullGcCount - g_fullGcCountSeenByFinalization > 0)
+        // we use unsigned math here as the collection counts, which are size_t internally,
+        // can in theory overflow an int and wrap around.
+        // unsigned math would have more defined/portable behavior in such case
+        if ((unsigned int)desiredFullGcCount - (unsigned int)g_fullGcCountSeenByFinalization > 0)
```

Comment:
Do you need to cast the result back to `int`?

`unsigned` minus `unsigned` is going to be unsigned. It means that the current version is same as `if (desiredFullGcCount != g_fullGcCountSeenByFinalization)`.

Reply:
Yes, I think I need a cast to `int`. The result of unsigned subtraction is indeed unsigned.

Reply:
Also, I think we see that in gcstress taking long time.

Reply:
It is relatively fast to run gcstress these days (about 1 - 1.5 hours), but in this PR it's been 2+ hours and nothing completed yet.